Add Provider VDC and VM Group data sources, VM Placement policies resource+data source and support for VM Placement policies in VDCs (#904)
vmware/terraform-provider-vcd: website/docs/d/provider_vdc.html.markdown

---
layout: "vcd"
page_title: "VMware Cloud Director: vcd_provider_vdc"
sidebar_current: "docs-vcd-data-source-provider-vdc"
description: |-

>   Provides an Provider VDC data source.

Reviewer: ```suggestion
  Provides a Provider VDC data source.
```

Author: Fixed, thanks for spotting that!

---

# vcd\_provider\_vdc

Provides a VMware Cloud Director Provider VDC data source. A Provider VDC can be used to reference a Provider VDC and use its 
data within other resources or data sources.

Supported in provider *v3.8+*

## Example Usage

```hcl
data "vcd_provider_vdc" "my-pvdc" {
  name = "my-pvdc"
}

output "provider_vdc" {
  value = data.vcd_provider_vdc.my-pvdc.id
}

```

## Argument Reference

The following arguments are supported:
 
* `name` - (Required) Provider VDC name

## Attribute reference

* `description` - Optional description of the Provider VDC.
* `status` - Status of the Provider VDC, it can be -1 (creation failed), 0 (not ready), 1 (ready), 2 (unknown) or 3 (unrecognized).
* `is_enabled` - True if this Provider VDC is enabled and can provide resources to organization VDCs. A Provider VDC is always enabled on creation.
* `capabilities` - Set of virtual hardware versions supported by this Provider VDC.

> * `compute_capacity` - Single-element list with an indicator of CPU and memory capacity. See [Compute Capacity](#compute-capacity) below for details.

Reviewer: ```suggestion
* `compute_capacity` - An indicator of CPU and memory capacity. See [Compute Capacity](#compute-capacity) below for details.
```

Author: Changed, thanks for the suggestion

* `highest_supported_hardware_version` - The highest virtual hardware version supported by this Provider VDC.
* `nsxt_manager_id` - ID of the registered NSX-T Manager that backs networking operations for this Provider VDC.
* `storage_containers_ids` - Set of IDs of the vSphere datastores backing this provider VDC.
* `external_network_ids` - Set of IDs of external networks.
* `storage_profile_ids` - Set of IDs to the storage profiles available to this Provider VDC.
* `resource_pool_ids` - Set of IDs of the resource pools backing this provider VDC.
* `network_pool_ids` - Set IDs of the network pools used by this Provider VDC.
* `universal_network_pool_id` - ID of the universal network reference.
* `host_ids` - Set with all the hosts which are connected to VC server.
* `vcenter_id` - ID of the vCenter server that provides the resource pools and datastores.
* `metadata` - Key and value pairs for Provider VDC metadata.

<a id="compute-capacity"></a>
## Compute Capacity

The `compute_capacity` attribute is a list with a single item which has the following nested attributes:

> * `cpu` - Single-element list with an indicator of CPU. See below.

Reviewer: ```suggestion
* `cpu` - An indicator of CPU. See [CPU and memory](#cpu-and-memory) below.
```

Author: Changed, thanks for the suggestion

> * `memory` - Single-element list with an indicator of memory. See below.

Reviewer: ```suggestion
* `memory` - `An indicator of memory. See [CPU and memory](#cpu-and-memory) below.
```

Author: Changed, thanks for the suggestion

* `is_elastic` -  True if compute capacity can grow or shrink based on demand.
* `is_ha` - True if compute capacity is highly available.

> 

Reviewer: ```suggestion

<a id="cpu-and-memory"></a>
### CPU and memory
```

Author: Added!

> The `cpu` and `memory` attributes are a list with a single item, that have the following nested attributes:

Reviewer: ```suggestion
The `cpu` and `memory` indicators have the following attributes:
```

Author: Changed, thanks for the suggestion

* `compute_provider_scope` - Represents the compute fault domain for this Provider VDC. This value is a tenant-facing tag that is shown to tenants when viewing fault domains of the child Organization VDCs (for example, a VDC Group).
* `highest_supported_hardware_version` - The highest virtual hardware version supported by this Provider VDC.
* `nsxt_manager_id` - ID of the registered NSX-T Manager that backs networking operations for this Provider VDC.
* `storage_containers_ids` - Set of IDs of the vSphere datastores backing this provider VDC.
* `external_network_ids` - Set of IDs of external networks.
* `storage_profile_ids` - Set of IDs to the storage profiles available to this Provider VDC.
* `resource_pool_ids` - Set of IDs of the resource pools backing this provider VDC.
* `network_pool_ids` - Set IDs of the network pools used by this Provider VDC.
* `universal_network_pool_id` - ID of the universal network reference.
* `host_ids` - Set with all the hosts which are connected to VC server.
* `vcenter_id` - ID of the vCenter server that provides the resource pools and datastores.

> * `metadata` - Key and value pairs for Provider VDC metadata.

Reviewer 2: What about capitalizing keywords above for easier reading and eye-scanning? E.g. External Networks, Storage Profiles, etc.

Author: Done!
